CMakeLists.txt

```diff
@@ -74,6 +74,67 @@ if ((NOT "${RELEASE_TYPE}" STREQUAL "PRODUCTION") AND (NOT "${RELEASE_TYPE}" STR
     set(RELEASE_TYPE "DEV")
 endif()
 
+# VIRCADIA_OPTIMIZE
+# Variable determining vircadia optimization. If not set, it defaults to true.
+# It's used to determine build flags for main codebase and for VCPKG dependencies.
+# Should be set to false to get completely unoptimized build for easier line-by-line debugging
+
+if( NOT WIN32 )
+    if(NOT DEFINED VIRCADIA_OPTIMIZE)
+        message("Enabling code optimization for Vircadia and compiled dependencies")
+        set(VIRCADIA_OPTIMIZE true CACHE BOOL "Enable code optimization for Vircadia and compiled dependencies")
+    endif()
+
+    #compiler needs to be detected before building VCPKG dependencies
+    set(CMAKE_PLATFORM_INFO_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+    include(CMakeDetermineCXXCompiler)
+
+    set(VIRCADIA_OPTIMIZE_FLAGS "")
+
+            set(VIRCADIA_OPTIMIZE_FLAGS "-O3 -fPIC")
+    if(VIRCADIA_OPTIMIZE)
+
+        file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/_env/VIRCADIA_OPTIMIZE.txt" "${VIRCADIA_OPTIMIZE}")
+        if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+            message("Clang compiler detected, adding -O3 -fPIC flags")
+        elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+            message("GCC compiler detected, adding -O3 -fPIC flags")
+            set(VIRCADIA_OPTIMIZE_FLAGS "-O3 -fPIC")
+        else()
+            message("No predefined optimization flags for compiler ${CMAKE_CXX_COMPILER_ID}")
+        endif()
+    endif()
+    MESSAGE(STATUS "VIRCADIA_OPTIMIZE: ${VIRCADIA_OPTIMIZE}")
+
+    # VIRCADIA_CPU_ARCHITECTURE
```

Comment:
I really don't like the VIRCADIA_CPU_ARCHITECTURE part. The person building has to somehow find out how to use this, or they will end up with "native" builds that aren't portable.
For example I have no idea how this works, and if I wouldn't have looked at this PR, I might have just ended up with broken release builds.
This should at least be mentioned in https://github.com/vircadia/vircadia/blob/master/BUILD.md#possible-environment-variables

Reply:
Thanks! It definitely has to be mentioned in BUILD.md document.  I will update this and look at other issue tomorrow morning.

Reply:
I added information about it to BUILD.md, BUILD_LINUX.md, BUILD_OSX.md

Reply:
My plan is to have vircadia-builder take this into account and automatically set more compatible optimization options for cases like AppImage

Reply:
I added information on how to disable VIRCADIA_CPU_ARCHITECTURE, and updated Github workflows accordingly